lib/fizzy/types.hpp

```diff
@@ -136,15 +136,15 @@ struct Memory
     Limits limits;
 };
 
-enum class GlobalInitType : uint8_t
+enum class ConstantInitType : uint8_t
```

Comment:
Could also scope this enum within `ConstantExpression`.

Reply:
Unless it is named like that in the spec, the `ConstantInitKind` is preferable name. In general, in language parsing context "kind" is better name than "type" for something that is not a type in language sense (`i32`, `i64`, etc).

Reply:
It is called type in the spec.

![Screenshot 2020-01-14 at 16 18 06](https://user-images.githubusercontent.com/20340/72361462-749dcd00-36e9-11ea-8238-d2698ceedd3a.png)

Reply:
Can we scope it within `ConsantExpression`? It is specific to it.

Reply:
And I suppose we rename `init_type` to `init_kind`.

Reply:
I would propose the following renaming:
```
ConstantExpression expression;
expression.kind = ConstantExpression::Kind::Constant;
expression.value.constant = 1;
```
and
```
ConstantExpression expression;
expression.kind = ConstantExpression::Kind::GlobalGet;
expression.value.global_index = 1;
```

Can we use anonymous unions or some compilers have issues with that?

Reply:
1. The proposal looks good.
2. The nested structs and enums can be annoying to use, but we can start with it anyway.
3. Anonymous unions are not C++ compliant. I believe only MSVC is complaining in some pedantic mode.
4. You can also check `std::variant` type.